File: money_exchange_rate/src/main/java/com/tools/money/ViewModelFactory.java
package com.tools.money;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

public class ViewModelFactory implements ViewModelProvider.Factory {

    ExchangeRateRepo exchangeRateRepo;

    public ViewModelFactory(ExchangeRateRepo exchangeRateRepo) {
        this.exchangeRateRepo = exchangeRateRepo;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new ExchangeRateViewModel(exchangeRateRepo);

    }
}

File: weather/src/main/java/com/tools/weather/ViewModelFactory.kt
package com.tools.weather

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class ViewModelFactory(private val weatherRepo: WeatherRepo) : ViewModelProvider.Factory {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(WeatherViewModel::class.java)) {
            return WeatherViewModel(weatherRepo) as T
        }

        throw IllegalArgumentException("Unknown class name")
    }

}